security/pkg/caclient/keycertbundlerotator.go

```diff
@@ -87,7 +87,12 @@ func (c *keyCertBundleRotatorImpl) Start(errCh chan<- error) {
 			}
 		}
 		log.Infof("Retrieve new key and certs.")
-		certBytes, certChainBytes, privateKeyBytes, err := c.retriever.Retrieve()
+		co, err := c.keycert.CertOptions()
+		if err != nil {
+			errCh <- fmt.Errorf("failed to extact CertOptions from bundle %v", err)
```

Comment:
Before return, you need to set:
c.stoppedMutex.Lock()
c.stopped = true
c.stoppedMutex.Unlock()

Reply:
Thanks! I add a defer to make sure we always do this clean-up when function exits.

Reply:
Great. Thanks!